src/components/ColorRampDropdown.tsx

```diff
@@ -106,22 +195,37 @@ const ColorRampSelector: React.FC<ColorRampSelectorProps> = (propsInput): ReactE
         <Tooltip
           // Force the tooltip to be hidden (open=false) when disabled
           open={props.disabled ? false : undefined}
-          title={selectedRampData.name}
+          title={props.useCategoricalPalettes ? selectedPaletteName : selectedRampData.name}
           placement="right"
           trigger={["focus", "hover"]}
         >
           <Button id={styles.selectorButton} disabled={props.disabled} onClick={() => setForceOpen(!forceOpen)}>
-            <img src={selectedRampColorUrl} />
+            {
+              // Switch between categorical palette and color map as needed
+              props.useCategoricalPalettes ? (
+                <div
+                  className={styles.categoricalColor}
+                  style={getColorPaletteStyling(props.selectedPalette.slice(0, props.numCategories))}
+                ></div>
+              ) : (
+                <img src={selectedRampColorUrl} />
```

Comment:
slightly surprised that you can't use the same visual representation for both!

Reply:
Hmm... I *could* make it work if I changed `ColorRamp` to support hard stop gradients, since currently it assumes all gradients have equally spaced color stops. Let me noodle on this one...

Reply:
I was able to make this work, though I need to do some more cleanup in the meantime and I'll mark this as a draft until it's ready to look at again.

Reply:
Done!